plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -412,8 +417,11 @@ else if (delete.content() == EQUALITY_DELETES) {
                         .map(id -> getColumnHandle(schema.findField(id), typeManager))
                         .collect(toImmutableList());
 
+                Schema deleteSchema = schemaFromHandles(columns);
+                StructLikeSet deleteSet = equalityDeleteSets.getDeleteSet(Sets.newHashSet(fieldIds), deleteSchema);
```

Comment:
I don't think that switching the List to a Set here is safe. If you have two eq delete files with the same columns but in a different order this will be a problem.

Reply:
Already replied above.